assets/src/editor/events/dates-and-times/editor-date/list-view/editor-date-entity-list-item.js

```diff
@@ -29,26 +28,33 @@ const { getBackgroundColorClass } = dateTimeModel;
  * @param {Object} date    JSON object defining the Event Date
  * @return {string}        The date rendered as a block
  */
-class EditorDateEntityListItem extends Component {
+const EditorDateEntityListItem = ( {
+	dateEntity,
+	doRefresh,
+} ) => {
+	const eventEntity = useEditorEventEntity();
 	/**
 	 * getStatusClass
 	 *
 	 * @function
 	 * @param {Object} dateEntity    JSON object defining the Event Date
 	 * @return {string}    CSS class corresponding to the Date status
 	 */
-	getStatusClass = ( dateEntity ) => {
-		switch ( dateEntity.status ) {
-			case 'DTA' :
-				return 'ee-datetime-active';
-			case 'DTE' :
-				return 'ee-datetime-expired';
-			case 'DTS' :
-				return 'ee-datetime-sold-out';
-			case 'DTU' :
-				return 'ee-datetime-upcoming';
-		}
-	};
+	const statusClass = useMemo(
+		() => {
+			switch ( dateEntity.status ) {
+				case 'DTA' :
+					return 'ee-datetime-active';
+				case 'DTE' :
+					return 'ee-datetime-expired';
+				case 'DTS' :
+					return 'ee-datetime-sold-out';
+				case 'DTU' :
+					return 'ee-datetime-upcoming';
+			}
+		},
+		[ dateEntity ]
```

Comment:
Would it not be better to simply use `dateEntity.status` for the dependency?
ie: it wouldn't be necessary to recalculate this if the date's name or other property changed

Reply:
Ya good point.  It also wouldn't be necessary to recalculate if the dateEntity changed, but the status was the same.

Reply:
oh HA! ya didn't even think of that!
ya sold out is sold out regardless of what date you're referring to.